appx manifest is invalid

## Current behavior

Recent builds of resizetizer may [fail the build](https://dev.azure.com/uno-platform/Uno%20Platform/_build/results?buildId=128154&view=logs&j=27c58675-674e-5cec-f5db-75e6ea252011&t=e076bf40-55f1-5179-da67-9a21e72ecb1c&l=402):

```
##[error]MakeAppx(0,0): Error : Error info: error C00CE015: App manifest validation error: The app manifest must be valid as per schema: Line 17, Column 140, Reason: The attribute '{[http://schemas.microsoft.com/appx/manifest/foundation/windows10}MaxVersionTested'](http://schemas.microsoft.com/appx/manifest/foundation/windows10%7DMaxVersionTested') on the element '{[http://schemas.microsoft.com/appx/manifest/foundation/windows10}TargetDeviceFamily'](http://schemas.microsoft.com/appx/manifest/foundation/windows10%7DTargetDeviceFamily') is not defined in the DTD/Schema.
MakeAppx : error : Error info: error C00CE015: App manifest validation error: The app manifest must be valid as per schema: Line 17, Column 140, Reason: The attribute '{[http://schemas.microsoft.com/appx/manifest/foundation/windows10}MaxVersionTested'](http://schemas.microsoft.com/appx/manifest/foundation/windows10%7DMaxVersionTested') on the element '{[http://schemas.microsoft.com/appx/manifest/foundation/windows10}TargetDeviceFamily'](http://schemas.microsoft.com/appx/manifest/foundation/windows10%7DTargetDeviceFamily') is not defined in the DTD/Schema. [D:\a\1\s\src\samples\WinUI\Uno.Themes.WinUI.Samples.Windows\Uno.Themes.WinUI.Samples.Windows.csproj]
##[error]MakeAppx(0,0): Error : Package creation failed.
MakeAppx : error : Package creation failed. [D:\a\1\s\src\samples\WinUI\Uno.Themes.WinUI.Samples.Windows\Uno.Themes.WinUI.Samples.Windows.csproj]
##[error]MakeAppx(0,0): Error : 0x80080204 - The specified package format is not valid: The package manifest is not valid.
MakeAppx : error : 0x80080204 - The specified package format is not valid: The package manifest is not valid. [D:\a\1\s\src\samples\WinUI\Uno.Themes.WinUI.Samples.Windows\Uno.Themes.WinUI.Samples.Windows.csproj]
##[error]Process 'msbuild.exe' exited with code '1'.
```

## Expected behavior

<!-- Describe what the desired behavior would be. -->

## How to reproduce it (as minimally and precisely as possible)

<!-- Please provide a **MINIMAL REPRO PROJECT** and the **STEPS TO REPRODUCE**-->

## Environment

<!-- For bug reports Check one or more of the following options with "x" -->

Nuget Package:

Package Version(s):

Affected platform(s):

- [ ] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renders for Xamarin.Forms
- [ ] Windows
- [ ] Build tasks

Visual Studio:

- [ ] 2017 (version: )
- [ ] 2019 (version: )
- [ ] for Mac (version: )

Relevant plugins:

- [ ] Resharper (version: )

## Anything else we need to know?

<!-- We would love to know of any friction, apart from knowledge, that prevented you from sending in a pull-request -->

--- a/src/Resizetizer/src/FileHelper.cs
+++ b/src/Resizetizer/src/FileHelper.cs
@@ -22,6 +22,12 @@ namespace Uno.Resizetizer
                 if (!File.Exists(fileName)
                     || !File.ReadAllText(fileName).Equals(File.ReadAllText(tempFile)))
                 {
+                    if (File.Exists(fileName))
+                    {
+                        log.LogMessage(MessageImportance.Low, $"Updating file: {fileName}");
+                        File.Delete(fileName);
+                    }
+
                     File.Move(tempFile, fileName);
                 }
                 else

--- a/src/Resizetizer/src/GeneratePackageAppxManifest.cs
+++ b/src/Resizetizer/src/GeneratePackageAppxManifest.cs
@@ -516,11 +516,26 @@ namespace Uno.Resizetizer
 
 		private static void SetVersion(XElement target, XName attributeName, string version)
 		{
-			var attr = target.Attribute(attributeName);
+			var attr = target.Attributes().FirstOrDefault(x => IsVersionAttribute(x, attributeName));
 			if (attr is null || string.IsNullOrEmpty(attr.Value) || attr.Value == PackageVersionPlaceholder)
 			{
 				target.SetAttributeValue(attributeName, version);
 			}
+		}
+
+		static bool IsVersionAttribute(XAttribute attribute, XName attributeName)
+		{
+			var currentAttributeName = attribute.Name.LocalName;
+			var expectedAttributeName = attributeName.LocalName;
+
+			var currentAttributeNamespace = attribute.Name.Namespace.NamespaceName;
+			var expectedAttributeNamespace = attributeName.NamespaceName;
+
+			// The Version may not have a current Namespace and should use the default namespace
+			if (string.IsNullOrEmpty(currentAttributeNamespace))
+				return currentAttributeName == expectedAttributeName;
+
+			return currentAttributeName == expectedAttributeName && currentAttributeNamespace == expectedAttributeNamespace;
 		}
 
 		public static bool TryMergeVersionNumbers(string? displayVersion, string? version, out string? finalVersion)

--- a/src/Resizetizer/test/UnitTests/GeneratePackageAppxManifestTests.cs
+++ b/src/Resizetizer/test/UnitTests/GeneratePackageAppxManifestTests.cs
@@ -1,4 +1,4 @@
-﻿#nullable enable
+#nullable enable
 using Microsoft.Build.Framework;
 using Microsoft.Build.Utilities;
 using System;
@@ -20,6 +20,7 @@ namespace Uno.Resizetizer.Tests
 			string? version = null,
 			string? displayName = null,
 			string? description = null,
+			string? applicationPublisher = null,
 			ITaskItem? appIcon = null,
 			ITaskItem? splashScreen = null)
 		{
@@ -35,6 +36,7 @@ namespace Uno.Resizetizer.Tests
 				ApplicationTitle = displayName,
 				AssemblyName = GetType().Assembly.FullName,
 				Description = description,
+				ApplicationPublisher = applicationPublisher,
 				AppIcon = appIcon == null ? null : new[] { appIcon },
 				SplashScreen = splashScreen == null ? null : new[] { splashScreen },
 				TargetFramework = "windows"
@@ -49,8 +51,11 @@ namespace Uno.Resizetizer.Tests
 		string? version = null,
 		string? displayName = null,
 		string? description = null,
+		string? applicationPublisher = null,
 		ITaskItem? appIcon = null,
-		ITaskItem? splashScreen = null)
+		ITaskItem? splashScreen = null,
+		string? targetPlatformVersion = null,
+		string? targetPlatformMinVersion = null)
 		{
 			return new()
 			{
@@ -63,10 +68,13 @@ namespace Uno.Resizetizer.Tests
 				ApplicationVersion = version,
 				ApplicationTitle = displayName,
 				Description = description,
-                AssemblyName = GetType().Assembly.FullName,
-                AppIcon = appIcon == null ? null : new[] { appIcon },
+				ApplicationPublisher = applicationPublisher,
+				AssemblyName = GetType().Assembly.GetName().Name,
+				AppIcon = appIcon == null ? null : new[] { appIcon },
 				SplashScreen = splashScreen == null ? null : new[] { splashScreen },
-				TargetFramework = "windows"
+				TargetFramework = "windows",
+				TargetPlatformVersion = targetPlatformVersion,
+				TargetPlatformMinVersion = targetPlatformMinVersion
 			};
 		}
 
@@ -96,7 +104,7 @@ namespace Uno.Resizetizer.Tests
 
 			var inputFilename = $"testdata/appxmanifest/typical.appxmanifest";
 			var task = GetNewTask(inputFilename,
-                applicationId: "com.contoso.myapp",
+				applicationId: "com.contoso.myapp",
 				displayVersion: "2.5",
 				version: "3",
 				displayName: "Fishy Things",
@@ -134,7 +142,7 @@ namespace Uno.Resizetizer.Tests
 
 			var inputFilename = $"testdata/appxmanifest/{input}.appxmanifest";
 			var task = GetNewTask(inputFilename,
-                applicationId: "com.contoso.myapp",
+				applicationId: "com.contoso.myapp",
 				displayVersion: "1.0.0",
 				version: "1",
 				displayName: "Sample App",
@@ -228,6 +236,66 @@ namespace Uno.Resizetizer.Tests
 
 			// Assert
 			Assert.True(LogWarningEvents.Count > 0, "Warnings should be greater than zero");
+		}
+
+		[Fact]
+		public void TaskShouldNotDuplicateTargetVersions()
+		{
+			// Arrange
+			var taskItem = new TaskItem("testdata/appxmanifest/duplicate-versions.input.appxmanifest");
+			var task = GetNewTask(
+				appxManifests: [taskItem],
+				targetPlatformVersion: "10.0.19041.0",
+				targetPlatformMinVersion: "10.0.17763.0");
+
+			// Act
+			task.Execute();
+
+			// Assert
+			AssertExpectedManifest(task.GeneratedAppxManifest.ItemSpec, taskItem.ItemSpec);
+		}
+
+		[Fact]
+		public void TaskShouldBeAbleToOverwriteAppxManifest()
+		{
+			// Arrange
+			var taskItem = new TaskItem("testdata/appxmanifest/empty.appxmanifest");
+			var task = GetNewTask(
+				appxManifests: [taskItem],
+				displayVersion: "1.0.0",
+				version: "1",
+				targetPlatformVersion: "10.0.19041.0",
+				targetPlatformMinVersion: "10.0.17763.0");
+
+			// Act
+			task.Execute();
+
+			// Assert
+			var generatedPath = task.GeneratedAppxManifest.ItemSpec;
+			var initial = XDocument.Load(generatedPath).ToString();
+			Assert.True(File.Exists(generatedPath));
+
+			// Act
+			for (var i = 2; i < 5; i++)
+			{
+				task.ApplicationVersion = $"{i}";
+				task.Execute();
+
+				// Assert
+				Assert.Equal(generatedPath, task.GeneratedAppxManifest.ItemSpec);
+				Assert.True(File.Exists(task.GeneratedAppxManifest.ItemSpec));
+				var updated = XDocument.Load(generatedPath).ToString();
+
+				Assert.NotEqual(initial, updated);
+			}
+		}
+
+		static void AssertExpectedManifest(string generatedManifestPath, string inputManifestPath)
+		{
+			var generated = XDocument.Load(generatedManifestPath).ToString();
+			var expected = XDocument.Load(inputManifestPath.Replace(".input.", ".expected.")).ToString();
+
+			Assert.Equal(expected, generated);
 		}
 	}
 }

--- a/src/Resizetizer/test/UnitTests/testdata/appxmanifest/duplicate-versions.expected.appxmanifest
+++ b/src/Resizetizer/test/UnitTests/testdata/appxmanifest/duplicate-versions.expected.appxmanifest
@@ -1,0 +1,28 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Package xmlns="http://schemas.microsoft.com/appx/manifest/foundation/windows10" xmlns:uap="http://schemas.microsoft.com/appx/manifest/uap/windows10" xmlns:rescap="http://schemas.microsoft.com/appx/manifest/foundation/windows10/restrictedcapabilities" IgnorableNamespaces="uap rescap">
+	<Identity Name="Uno.Themes.WinUI.Samples" Publisher="O=Uno.Themes.WinUI.Samples" Version="1.0.0.0" />
+	<Properties>
+		<DisplayName>Uno.Themes.WinUI.Samples</DisplayName>
+		<PublisherDisplayName>Uno.Themes.WinUI.Samples</PublisherDisplayName>
+	</Properties>
+	<Dependencies>
+		<TargetDeviceFamily Name="Windows.Universal" MinVersion="10.0.17763.0" MaxVersionTested="10.0.19041.0" />
+		<TargetDeviceFamily Name="Windows.Desktop" MinVersion="10.0.17763.0" MaxVersionTested="10.0.19041.0" />
+	</Dependencies>
+	<Resources>
+		<Resource Language="x-generate" />
+	</Resources>
+	<Applications>
+		<Application Id="App" Executable="$targetnametoken$.exe" EntryPoint="$targetentrypoint$">
+			<uap:VisualElements DisplayName="Uno.Themes.WinUI.Samples" Description="Uno.Themes.WinUI.Samples">
+				<uap:SplashScreen />
+				<uap:DefaultTile ShortName="Uno.Resizetizer.UnitTests">
+					<uap:ShowNameOnTiles />
+				</uap:DefaultTile>
+			</uap:VisualElements>
+		</Application>
+	</Applications>
+	<Capabilities>
+		<rescap:Capability Name="runFullTrust" />
+	</Capabilities>
+</Package>

--- a/src/Resizetizer/test/UnitTests/testdata/appxmanifest/duplicate-versions.input.appxmanifest
+++ b/src/Resizetizer/test/UnitTests/testdata/appxmanifest/duplicate-versions.input.appxmanifest
@@ -1,0 +1,43 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+
+<Package
+  xmlns="http://schemas.microsoft.com/appx/manifest/foundation/windows10"
+  xmlns:uap="http://schemas.microsoft.com/appx/manifest/uap/windows10"
+  xmlns:rescap="http://schemas.microsoft.com/appx/manifest/foundation/windows10/restrictedcapabilities"
+  IgnorableNamespaces="uap rescap">
+
+  <Identity
+    Name="Uno.Themes.WinUI.Samples"
+    Publisher="O=Uno.Themes.WinUI.Samples"
+    Version="1.0.0.0" />
+
+  <Properties>
+    <DisplayName>Uno.Themes.WinUI.Samples</DisplayName>
+    <PublisherDisplayName>Uno.Themes.WinUI.Samples</PublisherDisplayName>
+  </Properties>
+
+  <Dependencies>
+    <TargetDeviceFamily Name="Windows.Universal" MinVersion="10.0.17763.0" MaxVersionTested="10.0.19041.0" />
+    <TargetDeviceFamily Name="Windows.Desktop" MinVersion="10.0.17763.0" MaxVersionTested="10.0.19041.0" />
+  </Dependencies>
+
+  <Resources>
+    <Resource Language="x-generate"/>
+  </Resources>
+
+  <Applications>
+    <Application Id="App"
+      Executable="$targetnametoken$.exe"
+      EntryPoint="$targetentrypoint$">
+      <uap:VisualElements
+        DisplayName="Uno.Themes.WinUI.Samples"
+        Description="Uno.Themes.WinUI.Samples">
+        <uap:SplashScreen />
+      </uap:VisualElements>
+    </Application>
+  </Applications>
+
+  <Capabilities>
+    <rescap:Capability Name="runFullTrust" />
+  </Capabilities>
+</Package>

